extensions/lifecycle/LifecycleQueuePopulator.js

```diff
@@ -0,0 +1,136 @@
+        return this.zkClient.exists(path, (err, stat) => {
+            if (err) {
+                this.log.error('Could not check existence of zookeeper node', {
+                    method: 'LifecycleQueuePopulator._updateZookeeper',
+                    zkPath: path,
+                    error: err,
+                });
+                return undefined;
+            }
+            // Remove existing node if deleting the bucket or its configuration.
+            if (stat &&
+            (value.deleted || value.lifecycleConfiguration === null)) {
+                return this.zkClient.remove(path, err => {
+                    if (err) {
+                        this.log.error('Could not remove zookeeper node', {
+                            method: 'LifecycleQueuePopulator._updateZookeeper',
+                            zkPath: path,
+                            error: err,
+                        });
+                    }
+                    return undefined;
+                });
+            }
+            if (value.lifecycleConfiguration !== null) {
+                return this.zkClient.create(path, err => {
+                    if (err) {
+                        this.log.error('Could not create new zookeeper node', {
+                            method: 'LifecycleQueuePopulator._updateZookeeper',
+                            zkPath: path,
+                            error: err,
+                        });
+                    }
+                    return undefined;
+                });
+            }
+            return undefined;
+        });
+    }
+
+    /**
+     * Filter record log entries for those that are potentially relevant to
+     * lifecycle.
+     * @param {Object} entry - The record log entry from metadata.
+     * @return {undefined}
+     */
+    filter(entry) {
+        if (entry.key.startsWith(mpuBucketPrefix) ||
+            entry.bucket === usersBucket) {
+            return undefined;
+        }
+        if (entry.type === 'put') {
+            return this._setup(err => {
+                if (err) {
+                    this.log.error('Could set up zookeeper path', {
+                        error: err,
+                    });
+                    return undefined;
+                }
+                return this._updateZookeeper(entry);
+            });
+        }
+        return undefined;
+    }
+
+    /**
+     * Set the zookeeper client.
+     * @param {Object} zkClient - The Zookeeper client instance.
+     * @return {LifecycleQueuePopulator} - The current LifecycleQueuePopulator
+     * instance.
+     */
+    setZookeeper(zkClient) {
+        this.zkClient = zkClient;
+        return this;
+    }
+
+    /**
+     * Set the zookeeper basepath for the populator.
+     * @param {object} extensions - The extensions configuration object.
+     * @return {LifecycleQueuePopulator} - The current LifecycleQueuePopulator
+     * instance.
+     */
+    setZookeeperPath(extensions) {
```

Comment:
I think there's more value in adding a `setConfig(backbeatConfig, extConfig)` method called from the log reader, mapping the `configValidator` model, where the extension can save any piece of configuration it deems useful to be used later for private matters, instead of one specific piece of it.

Reply:
Updated.